Module-4/Homework.playground/Contents.swift

```diff
@@ -1,2 +1,220 @@
+ 
+ You can set the values of the arrays above using whatever method you find the easiest. In previous modules, you were introduced to ranges and sequences in Swift. Leveraging those in the Array initializer will allow you to create the requested arrays in a single line. Don’t let the last two break your stride.
+ 
+ - Create a function called `evenNumbersArray` that takes a parameter of `[Int]` (array of `Int`) and returns `[Int]`. The array of `Int` returned should contain all the even numbers in the array passed. Call the function passing the `nums` array and print the output.
+ 
+ - Create a function called `sumOfArray` that takes a parameter of `[Int?]` and returns an `Int`. The function should return the sum of the array values passed that are not `nil`. Call the function passing the `numsWithNil` array, and print out the results.
+ 
+ - Create a function called `commonElementsSet` that takes two parameters of `[Int]` and returns a `Set<Int>` (set of `Int`). The function will return a `Set<Int>` of the values in both arrays.
+ 
+ - Call the function `commonElementsSet` passing the arrays `numsBy2`, `numsBy4`, and print out the results.
+ */
+let nums = Array(0...100)
+let numsWithNil = [79, nil, 80, nil, 90, nil, 100, 72]
+let numsBy2 = Array(stride(from: 2, through: 100, by: 2))
+let numsBy4 = Array(stride(from: 2, through: 100, by: 4))
+
+func evenNumbersArray(_ array: [Int]) -> [Int] {
+    array.filter { $0 % 2 == 0 }
+}
+
+func sumOfArray(_ array: [Int?]) -> Int {
+    array.reduce(0) { runningTotal, element in
+        runningTotal + (element ?? 0)
+    }
+}
+
+print(sumOfArray(numsWithNil))
+
+func commonElementsSet(_ arrayA: [Int], _ arrayB: [Int]) -> Set<Int> {
+    Set(arrayA).intersection(arrayB)
+}
+
+print(commonElementsSet(numsBy2, numsBy4))
+/*:
+ d) Create a struct called `Square` that has a stored property called `sideLength` and a computed property called `area`. Create an instance of `Square` and print out the area.
+*/
+struct Square {
+    var sideLength: Double
+    var area: Double { pow(sideLength, 2) }
+}
+
+let square = Square(sideLength: 5.0)
+print("square area:", square.area)
+/*: ## Part 3 - Above and Beyond.
+ */
+/*:
+ Create a `protocol` called `Shape` with a `calculateArea() -> Double` method. Create two `struct`s called `Circle` and `Rectangle` that conform to the protocol `Shape`. Both `Circle` and `Rectangle` should have appropriate stored properties for calculating the area.
+ 
+ Create instances of `Circle` and `Rectangle` and print out the area for each.
+ 
+ Next, extend the protocol `Shape` to add a new method called `calculateVolume() -> Double`.
+ 
+ Finally, create a `struct` called `Sphere` that conforms to `Shape`. `Sphere` should have appropriate stored properties for calculating area and volume.
+ 
+ Create an instance of `Sphere` and print out the area and volume.
+ */
+protocol Shape {
+    func calculateArea() -> Double
+    func calculateVolume() -> Double
```

Comment:
You don't actually need to declare the function here when you do it in an extension.